PowerEditor/src/WinControls/VerticalFileSwitcher/VerticalFileSwitcher.cpp

```diff
@@ -65,6 +68,129 @@ void VerticalFileSwitcher::startColumnSort()
 	updateHeaderArrow();
 }
 
+LRESULT VerticalFileSwitcher::listViewNotifyCustomDraw(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
+{
+	auto lplvcd = reinterpret_cast<LPNMLVCUSTOMDRAW>(lParam);
+
+	switch (lplvcd->nmcd.dwDrawStage)
+	{
+		case CDDS_PREPAINT:
+		{
+			if ((lplvcd->dwItemType == LVCDI_GROUP) && NppDarkMode::isThemeDark())
+			{
+				RECT rcHeader{};
+				ListView_GetGroupRect(lplvcd->nmcd.hdr.hwndFrom, lplvcd->nmcd.dwItemSpec, LVGGR_HEADER, &rcHeader);
+
+				HBRUSH hBrush = ::CreateSolidBrush(VerticalFileSwitcher::_bgColor);
+				::FillRect(lplvcd->nmcd.hdc, &rcHeader, hBrush);
+				::DeleteObject(hBrush);
+				hBrush = nullptr;
+			}
+			return CDRF_NOTIFYITEMDRAW;
+		}
+
+		case CDDS_ITEMPREPAINT:
+		{
+			const auto isSelected = ListView_GetItemState(lplvcd->nmcd.hdr.hwndFrom, lplvcd->nmcd.dwItemSpec, LVIS_SELECTED) == LVIS_SELECTED;
+			const bool isHot = (lplvcd->nmcd.uItemState & CDIS_HOT) == CDIS_HOT;
+			const bool isThemeDark = NppDarkMode::isThemeDark();
+			const int colorID = reinterpret_cast<TaskLstFnStatus*>(lplvcd->nmcd.lItemlParam)->_docColor;
+
+			if (colorID != -1)
+			{
+				if (isThemeDark)
+				{
+					lplvcd->clrText = NppDarkMode::getTextColor();
+				}
+
+				const COLORREF color = NppDarkMode::getIndividualTabColour(colorID, isThemeDark, false);
+				lplvcd->clrTextBk = color;
+
+				HBRUSH hBrush = ::CreateSolidBrush(color);
+				::FillRect(lplvcd->nmcd.hdc, &lplvcd->nmcd.rc, hBrush);
+				::DeleteObject(hBrush);
+				hBrush = nullptr;
+			}
+			else if (isThemeDark)
+			{
+				if (isSelected)
+				{
+					lplvcd->clrText = NppDarkMode::getTextColor();
+					lplvcd->clrTextBk = NppDarkMode::getSofterBackgroundColor();
+					::FillRect(lplvcd->nmcd.hdc, &lplvcd->nmcd.rc, NppDarkMode::getSofterBackgroundBrush());
+				}
+				else if (isHot)
+				{
+					lplvcd->clrText = NppDarkMode::getTextColor();
+					lplvcd->clrTextBk = NppDarkMode::getHotBackgroundColor();
+					::FillRect(lplvcd->nmcd.hdc, &lplvcd->nmcd.rc, NppDarkMode::getHotBackgroundBrush());
+				}
+			}
+
+			if (isSelected)
+			{
+				::DrawFocusRect(lplvcd->nmcd.hdc, &lplvcd->nmcd.rc);
+			} 
+			else if (isHot)
+			{
+				::FrameRect(lplvcd->nmcd.hdc, &lplvcd->nmcd.rc, isThemeDark ? NppDarkMode::getHotEdgeBrush() : ::GetSysColorBrush(COLOR_WINDOWTEXT));
+			}
+
+			return CDRF_NEWFONT;
+		}
+
+		default:
+			break;
+	}
+	return ::DefSubclassProc(hWnd, uMsg, wParam, lParam);
+}
+
+LRESULT CALLBACK VerticalFileSwitcher::FileSwitcherNotifySubclass(
+	HWND hWnd,
+	UINT uMsg,
+	WPARAM wParam,
+	LPARAM lParam,
+	UINT_PTR uIdSubclass,
+	DWORD_PTR /*dwRefData*/
+)
+{
+	switch (uMsg)
+	{
+		case WM_NCDESTROY:
+		{
+			::RemoveWindowSubclass(hWnd, VerticalFileSwitcher::FileSwitcherNotifySubclass, uIdSubclass);
+			break;
+		}
+
+		case WM_NOTIFY:
+		{
+			auto nmhdr = reinterpret_cast<LPNMHDR>(lParam);
+			switch (nmhdr->code)
+			{
+				case NM_CUSTOMDRAW:
+				{
+					constexpr size_t classNameLen = 16;
+					TCHAR className[classNameLen]{};

```

Comment:
Should it be `{}` or `{ '\0' }`?

Reply:
I usually use ` = { '\0' }` for "char" array, But it can be initialized with `{}`,`= {}`, ...